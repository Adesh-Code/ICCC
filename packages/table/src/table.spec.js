import React from "react";
import Table from "./table";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

const arrange = (tableRowProps) => {
  const utils = render(
    <Table {...tableRowProps}>
      <tbody>
        <tr>
          <td>Foo</td>
        </tr>
      </tbody>
    </Table>
  );
  return {
    ...utils,
    table: utils.container.firstChild.firstChild
  };
};

describe("Table", () => {
  it("renders", () => {
    const { table } = arrange();
    expect(table).toMatchSnapshot();
  });

  it("accepts classnames", () => {
    const { table } = arrange({ className: "bar" });
    expect(table).toHaveClass("bar");
  });

  it("renders loud palette", () => {
    const { table } = arrange({ loud: true });
    expect(table).toHaveClass("DhigTable--loud");
  });

  it("renders zebra styling", () => {
    const { table } = arrange({ zebra: true });
    expect(table).toHaveClass("DhigTable--zebra");
  });

  it("renders bordered variant", () => {
    const { table } = arrange({ bordered: true });
    expect(table).toHaveClass("DhigTable--bordered");
  });
});
