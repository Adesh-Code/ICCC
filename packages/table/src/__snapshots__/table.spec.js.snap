// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders 1`] = `
<table
  class="MuiTable-root DhigTable--verticalAlignment--top emotion-0"
>
  <tbody>
    <tr>
      <td>
        Foo
      </td>
    </tr>
  </tbody>
</table>
`;
